src/sycl/sycl_engine.hpp

```diff
@@ -61,12 +74,15 @@ inline std::vector<::sycl::device> get_sycl_devices(
     const uint32_t vendor_id
             = ((dev_type == ::sycl::info::device_type::gpu) ? 0x10DE
                                                             : intel_vendor_id);
+#elif defined(DNNL_SYCL_HIP)
+    const uint32_t vendor_id
+            = ((dev_type == ::sycl::info::device_type::gpu) ? 0x1002
```

Comment:
While `0x1002` seems to the correct vendor ID for AMD devices `get_info<:sycl::info::device::vendor_id>()` returns `10DE` (NVIDIA vendor ID) when HIP backend is used.
DPC++ compiler uses OpenCL backend by default therefore you need to set environment variable `SYCL_DEVICE_FILTER=hip` to enable HIP backend as the default one. When you do that `10DE` will be returned and engine won't be created. 
Since engine cannot be created I expect the rest of the code not to be functional.

Regarding vendor ID, I created an issue on DPC++ compiler github: https://github.com/intel/llvm/issues/6123